handle_multi_pay_invoice_request.go

```diff
@@ -0,0 +1,197 @@
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"strings"
+	"sync"
+
+	"github.com/nbd-wtf/go-nostr"
+	decodepay "github.com/nbd-wtf/ln-decodepay"
+	"github.com/sirupsen/logrus"
+)
+
+func (svc *Service) HandleMultiPayInvoiceEvent(ctx context.Context, sub *nostr.Subscription, request *Nip47Request, event *nostr.Event, app App, ss []byte) {
+
+	nostrEvent := NostrEvent{App: app, NostrId: event.ID, Content: event.Content, State: "received"}
+	err := svc.db.Create(&nostrEvent).Error
+	if err != nil {
+		svc.Logger.WithFields(logrus.Fields{
+			"eventId":   event.ID,
+			"eventKind": event.Kind,
+			"appId":     app.ID,
+		}).Errorf("Failed to save nostr event: %v", err)
+		svc.Logger.WithFields(logrus.Fields{
+			"eventId":   event.ID,
+			"eventKind": event.Kind,
+		}).Errorf("Failed to process event: %v", err)
+		return
+	}
+
+	multiPayParams := &Nip47MultiPayParams{}
+	err = json.Unmarshal(request.Params, multiPayParams)
+	if err != nil {
+		svc.Logger.WithFields(logrus.Fields{
+			"eventId":   event.ID,
+			"eventKind": event.Kind,
+			"appId":     app.ID,
+		}).Errorf("Failed to decode nostr event: %v", err)
+		svc.Logger.WithFields(logrus.Fields{
+			"eventId":   event.ID,
+			"eventKind": event.Kind,
+		}).Errorf("Failed to process event: %v", err)
+		return
+	}
+
+	var wg sync.WaitGroup
+	for _, invoiceInfo := range multiPayParams.Invoices {
+		wg.Add(1)
+		go func(invoiceInfo InvoiceInfo) {
+			defer wg.Done()
+			bolt11 := invoiceInfo.Invoice
+			// Convert invoice to lowercase string
+			bolt11 = strings.ToLower(bolt11)
+			paymentRequest, err := decodepay.Decodepay(bolt11)
+			if err != nil {
+				svc.Logger.WithFields(logrus.Fields{
+					"eventId":   event.ID,
+					"eventKind": event.Kind,
+					"appId":     app.ID,
+					"bolt11":    bolt11,
+				}).Errorf("Failed to decode bolt11 invoice: %v", err)
+
+				resp, err := svc.createResponse(event, Nip47Response{
+					ResultType: NIP_47_PAY_INVOICE_METHOD,
+					Error: &Nip47Error{
+						Code:    NIP_47_ERROR_INTERNAL,
+						Message: fmt.Sprintf("Failed to decode bolt11 invoice: %s", err.Error()),
+					},
+				}, ss)
+				if err != nil {
+					svc.Logger.WithFields(logrus.Fields{
+						"eventId":        event.ID,
+						"eventKind":      event.Kind,
+						"paymentRequest": invoiceInfo.Invoice,
+						"invoiceId":      invoiceInfo.Id,
+					}).Errorf("Failed to process event: %v", err)
+					return
+				}
+				dTag := []string{"a", fmt.Sprintf("%d:%s:%d", NIP_47_RESPONSE_KIND, event.PubKey, invoiceInfo.Id)}
+				resp.Tags = append(resp.Tags, dTag)
+				svc.PublishEvent(ctx, sub, event, resp)
+				return
+			}
+
+			hasPermission, code, message := svc.hasPermission(&app, event, request.Method, paymentRequest.MSatoshi)
```

Comment:
This could possibly bypass limits due to being sent in parallel (it's already an issue because we only check payments with preimage set. CC @bumi )